packages/cli/src/zosjobs/submit/Submit.shared.handler.ts

```diff
@@ -31,150 +31,158 @@ export default class SharedSubmitHandler extends ZosmfBaseHandler {
-            this.data.setObj(apiObj);
-
-            // Print data from spool content
-        } else {
-            for (const spoolFile of spoolFilesResponse) {
-                if (spoolFile.procName != null && spoolFile.procName.length > 0) {
-                    this.console.log("Spool file: %s (ID #%d, Step: %s, ProcStep: %s)",
-                        spoolFile.ddName, spoolFile.id, spoolFile.stepName, spoolFile.procName);
-                } else {
-                    this.console.log("Spool file: %s (ID #%d, Step: %s)",
-                        spoolFile.ddName, spoolFile.id, spoolFile.stepName);
+            // Print the response to the command
+            if (spoolFilesResponse == null) {
+                params.response.format.output({
+                    fields: ["jobid", "retcode", "jobname", "status"],
+                    output: apiObj,
+                    format: "object"
+                });
+                // Set the API object to the correct
+                this.data.setObj(apiObj);
+
+                // Print data from spool content
+            } else {
+                for (const spoolFile of spoolFilesResponse) {
+                    if (spoolFile.procName != null && spoolFile.procName.length > 0) {
+                        this.console.log("Spool file: %s (ID #%d, Step: %s, ProcStep: %s)",
+                            spoolFile.ddName, spoolFile.id, spoolFile.stepName, spoolFile.procName);
+                    } else {
+                        this.console.log("Spool file: %s (ID #%d, Step: %s)",
+                            spoolFile.ddName, spoolFile.id, spoolFile.stepName);
+                    }
+                    this.console.log(spoolFile.data);
                 }
-                this.console.log(spoolFile.data);
-            }
 
-            // Set the API object to the correct
-            this.data.setObj(spoolFilesResponse);
-        }
+                // Set the API object to the correct
+                this.data.setObj(spoolFilesResponse);
+            }
 
-        // Print path where spool content was downloaded
-        if (directory != null && spoolFilesResponse == null) {
-            directory = directory.includes("./") ? directory : `./${directory}`;
-            params.response.console.log(`Successfully downloaded output to ${directory}/${apiObj.jobid}`);
+            // Print path where spool content was downloaded
+            if (directory != null && spoolFilesResponse == null) {
+                directory = directory.includes("./") ? directory : `./${directory}`;
+                params.response.console.log(`Successfully downloaded output to ${directory}/${apiObj.jobid}`);
+            }
+            params.response.progress.endBar();
+            this.data.setMessage(`Submitted JCL contained in "${sourceType}": "${source}"`);
+        }catch (err){
+            throw new ImperativeError({msg: err?.mMessage?.message ?? err?.message ?? err?.stderr, causeErrors: err?.causeErrors});
```

Comment:
If the error is already an instance of ImperativeError, it's probably good to avoid "recasting" it so we don't lose any useful info stored on the object.

To avoid this, I think inside the catch block you could use code similar to what @awharn recently added in the DB2 plugin [here](https://github.com/zowe/zowe-cli-db2-plugin/blob/2ba3155f9e00b2304e5159b82559d2392ff2ae23/src/cli/DB2BaseHandler.ts#L38-L44):
```
if (err instanceof ImperativeError){
    throw err;
} else {
    throw new ImperativeError({msg: err.message, causeErrors: err});
}
```

Reply:
good suggestions thank you. these suggestions maintained the diagnostic information: 
![image](https://github.com/zowe/zowe-cli/assets/112635587/96cb3be6-f2ce-4254-bf06-238bd438a155)